src/SAWScript/Crucible/JVM/Override.hs

```diff
@@ -829,33 +806,30 @@ executePointsTo opts sc cc spec pt = do
   globals <- OM (use overrideGlobals)
   case pt of
 
-    JVMPointsToField loc ptr fid val ->
+    JVMPointsToField _loc ptr fid val ->
       do (_, val') <- resolveSetupValueJVM opts cc sc spec val
-         (_, ptr') <- resolveSetupValueJVM opts cc sc spec ptr
-         rval <- asRVal loc ptr'
+         rval <- resolveAllocIndexJVM ptr
          let dyn = injectJVMVal sym val'
          -- TODO: Change type of CJ.doFieldStore to take a FieldId instead of a String.
          -- Then we won't have to match the definition of 'fieldIdText' here.
          let key = J.unClassName (J.fieldIdClass fid) ++ "." ++ J.fieldIdName fid
          globals' <- liftIO $ CJ.doFieldStore sym globals rval key dyn
          OM (overrideGlobals .= globals')
 
-    JVMPointsToElem loc ptr idx val ->
+    JVMPointsToElem _loc ptr idx val ->
       do (_, val') <- resolveSetupValueJVM opts cc sc spec val
-         (_, ptr') <- resolveSetupValueJVM opts cc sc spec ptr
-         rval <- asRVal loc ptr'
+         rval <- resolveAllocIndexJVM ptr
          let dyn = injectJVMVal sym val'
          globals' <- liftIO $ CJ.doArrayStore sym globals rval idx dyn
          OM (overrideGlobals .= globals')
 
-    JVMPointsToArray loc ptr tt ->
+    JVMPointsToArray _loc ptr tt ->
       do (_ety, tts) <-
            liftIO (destVecTypedTerm sc tt) >>=
            \case
              Nothing -> fail "jvm_array_is: not a monomorphic sequence type"
```

Comment:
Is this failure hard to include in the structured error type you created?

Reply:
The intention of the `JVMSetupError` type is that it's for errors thrown in the `JVMSetupM` monad. This code is in the `OverrideMatcher` monad, which runs in a later phase (during symbolic simulation). That's why I treated it as out-of-scope for this PR; I'll overhaul the `OverrideMatcher` error handling in a separate PR.

Actually, because the types are checked in the earlier `JVMSetup` phase, we should be able to safely assume that the types will be correct at this point, so really this occurrence of `fail` should be upgraded to `panic`.

Reply:
Makes sense. I figured it was something like that.